File: IMASample/IMASample/Video.m
#import "Video.h"

@implementation Video
    
- (instancetype)initWithTitle:(NSString *)title
                    thumbnail:(UIImage *)thumbnail
                        video:(NSString *)video
                          tag:(NSString *)tag
                  adsResponse:(NSString *)adsResponse {
    self = [super init];
    if (self) {
        _title = [title copy];
        _thumbnail = [thumbnail copy];
        _video = [video copy];
        _tag = [tag copy];
        _adsResponse = [adsResponse copy];
    }
    return self;
}
    
@end

File: IMASample/IMASample-swift/Video.swift
//
//  Video.swift
//  IMASample-swift
//
//  Created by Vadik on 07/11/2017.
//  Copyright © 2017 kaltura. All rights reserved.
//

import UIKit

public class Video {
    var title: String
    var thumbnail: UIImage
    var tag: String?
    var adsResponse: String?
    
    init(title: String, thumbnail: UIImage, tag: String? = "", adsResponse: String? = "") {
        self.title = title
        self.thumbnail = thumbnail
        self.tag = tag
        self.adsResponse = adsResponse
    }
}
